app/services/find_or_register_user.rb

```diff
@@ -0,0 +1,27 @@
+class FindOrRegisterUser
+  attr_reader :github_id
+
+  def initialize(params)
+    @github_id = params[:github_id]
+  end
+
+  def perform
+    @_user = found_user = find_user
+
+    if found_user.nil?
+      @_user = register_user
+    end
+
+    @_user
+  end
+
+  private
+
+  def find_user
+    User.where(github_id: github_id).first
+  end
+
+  def register_user
```

Comment:
thoughts on naming this something like `CreateUser` since that is what we are doing in that class?